app/cdap/components/DirectiveInput/Components/InputPanel/index.tsx

```diff
@@ -0,0 +1,222 @@
+    event.preventDefault();
+    if (selectedIndex !== 0) {
+      setSelectedIndex(selectedIndex - 1);
+    }
+  };
+
+  // Used for navigating below in search list
+  const handleDownArrow = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (selectedIndex !== searchResults.length - 1) {
+      setSelectedIndex(selectedIndex + 1);
+    }
+  };
+
+  // Used for selecting item in search list
+  const handleEnterKey = () => {
+    if (inputText.length > 0) {
+      if (searchResults[selectedIndex]) {
+        handleListItemClick(searchResults[selectedIndex]);
+      } else {
+        onSearchItemClick(inputText);
+      }
+    }
+  };
+
+  // Used for filling input with active item in search list
+  const handleTabKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.preventDefault();
+    if (inputText.length === 0 || inputText.split(' ').length !== 1) {
+      return;
+    }
+    handleEnterKey();
+  };
+
+  useEffect(() => {
+    searchMatch(inputDirective);
+    setInputText(inputDirective);
+  }, [inputDirective]);
+
+  // Function called on every change in input to change search list to most appropriate matched list
+  const searchMatch = (searchString: string) => {
+    let searchList = [];
+    const spaceIndex = searchString.includes(' '); // As soon as directive is entered, we need column list to appear hence we are checking if space is present in it,
+    if (fuse && searchString.length > 0) {
+      if (!isDirectiveSet) {
+        searchList = fuse
+          .search(searchString)
+          .slice(0, 3)
+          .map((searchItem) => {
+            searchItem.uniqueId = uuidV4();
+            return searchItem;
+          });
+        reverse(searchList);
+      } else {
+        const characterToSearch = getLastWordOfSearchItem(searchString);
+        searchList = fuse.search(characterToSearch).map((searchItem) => {
+          searchItem.uniqueId = uuidV4();
+          return searchItem;
+        });
+        reverse(searchList);
+      }
+    }
+    setSearchResults(searchList);
+    setInputText(searchString);
+    setSelectedIndex(searchList.length - 1);
+    if (!isDirectiveSet) {
+      getDirectiveSyntax(searchList, spaceIndex);
+    }
+  };
+
+  // Function called when item is clicked in search list
+  const handleListItemClick = (listItem) => {
+    if (!isDirectiveSet) {
+      onSearchItemClick(listItem.item.directive);
+      getDirectiveSyntax([listItem], true);
+    } else {
+      const formattedString = getFormattedSyntax(inputText, listItem.item.label);
+      setInputText(formattedString);
+      onSearchItemClick(formattedString);
+    }
+  };
+
+  return (
+    <SearchWrapper data-testid="input-panel-wraper">
+      {searchResults.map((searchItem, searchItemIndex) =>
+        searchItemIndex === selectedIndex ? (
+          <ActiveResultRow
+            key={searchItem.uniqueId}
+            onClick={() => handleListItemClick(searchItem)}
```

Comment:
this is a box it shouldn't have an onclick handler on it

Reply:
done.